### ISSUE TITLE
Step - 05 - Build a Spring Boot microservice using Spring Cloud features not working

### ISSUE THREAD

**Issue by lalitkumarms** [microsoft/azure-spring-cloud-training#61]
Hi,

I tried using the step 5 to run on local as well as on azure spring cloud, but the endpoint gives error....

Whitelabel Error Page
This application has no explicit mapping for /error, so you are seeing this as a fallback.

Fri Oct 30 06:41:51 GMT 2020
There was an unexpected error (type=Not Found, status=404).

Here are the logs from cloud...

azureuser@springdevmachine:~$ az spring-cloud app logs --name spring-cloud-microservice -f
  .   ____          _            __ _ _
 /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
 \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
  '  |____| .__|_| |_|_| |_\__, | / / / /
 =========|_|==============|___/=/_/_/_/
 :: Spring Boot ::        (v2.3.2.RELEASE)

2020-10-30 06:39:33.877  INFO 1 --- [           main] c.c.c.ConfigServicePropertySourceLocator : Fetching config from server at : https://azure-spring-cloud-lab-lalit.svc.azuremicroservices.io/config
2020-10-30 06:39:36.444  INFO 1 --- [           main] c.c.c.ConfigServicePropertySourceLocator : Located environment: name=spring-cloud-microservice, profiles=[default], label=null, version=3a23e106c844e1e5980fc2becc841b59d94f268b, state=null
2020-10-30 06:39:36.446  INFO 1 --- [           main] b.c.PropertySourceBootstrapConfiguration : Located property source: [BootstrapPropertySource {name='bootstrapProperties-configClient'}, BootstrapPropertySource {name='bootstrapProperties-https://github.com/Azure-Samples/spring-cloud-sample-public-config.git/application.yml'}]
2020-10-30 06:39:36.456  INFO 1 --- [           main] com.example.demo.DemoApplication         : No active profile set, falling back to default profiles: default
2020-10-30 06:39:39.766  INFO 1 --- [           main] o.s.cloud.context.scope.GenericScope     : BeanFactory id=b9a90cf8-f8a5-385d-b0ab-3b28be03bdc3
2020-10-30 06:39:41.300  INFO 1 --- [           main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat initialized with port(s): 1025 (http)
2020-10-30 06:39:42.041  INFO 1 --- [           main] o.apache.catalina.core.StandardService   : Starting service [Tomcat]
2020-10-30 06:39:42.041  INFO 1 --- [           main] org.apache.catalina.core.StandardEngine  : Starting Servlet engine: [Apache Tomcat/9.0.37]
2020-10-30 06:39:42.356  INFO 1 --- [           main] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring embedded WebApplicationContext
2020-10-30 06:39:42.356  INFO 1 --- [           main] w.s.c.ServletWebServerApplicationContext : Root WebApplicationContext: initialization completed in 5804 ms
2020-10-30 06:39:42.589  WARN 1 --- [           main] c.n.c.sources.URLConfigurationSource     : No URLs will be polled as dynamic configuration sources.
2020-10-30 06:39:42.590  INFO 1 --- [           main] c.n.c.sources.URLConfigurationSource     : To enable URLs as dynamic configuration sources, define System property archaius.configurationSource.additionalUrls or make config.properties available on classpath.
2020-10-30 06:39:42.645  WARN 1 --- [           main] c.n.c.sources.URLConfigurationSource     : No URLs will be polled as dynamic configuration sources.
2020-10-30 06:39:42.645  INFO 1 --- [           main] c.n.c.sources.URLConfigurationSource     : To enable URLs as dynamic configuration sources, define System property archaius.configurationSource.additionalUrls or make config.properties available on classpath.
2020-10-30 06:39:44.677  INFO 1 --- [           main] o.s.s.concurrent.ThreadPoolTaskExecutor  : Initializing ExecutorService 'applicationTaskExecutor'
2020-10-30 06:39:45.544  INFO 1 --- [           main] DiscoveryClientOptionalArgsConfiguration : Eureka HTTP Client uses Jersey
2020-10-30 06:39:45.867  WARN 1 --- [           main] ockingLoadBalancerClientRibbonWarnLogger : You already have RibbonLoadBalancerClient on your classpath. It will be used by default. As Spring Cloud Ribbon is in maintenance mode. We recommend switching to BlockingLoadBalancerClient instead. In order to use it, set the value of `spring.cloud.loadbalancer.ribbon.enabled` to `false` or remove spring-cloud-starter-netflix-ribbon from your project.
2020-10-30 06:39:46.355  INFO 1 --- [           main] o.s.c.n.eureka.InstanceInfoFactory       : Setting initial instance status as: STARTING
2020-10-30 06:39:46.636  INFO 1 --- [           main] com.netflix.discovery.DiscoveryClient    : Initializing Eureka in region us-east-1
2020-10-30 06:39:47.167  INFO 1 --- [           main] c.n.d.provider.DiscoveryJerseyProvider   : Using JSON encoding codec LegacyJacksonJson
2020-10-30 06:39:47.168  INFO 1 --- [           main] c.n.d.provider.DiscoveryJerseyProvider   : Using JSON decoding codec LegacyJacksonJson
2020-10-30 06:39:47.642  INFO 1 --- [           main] c.n.d.provider.DiscoveryJerseyProvider   : Using XML encoding codec XStreamXml
2020-10-30 06:39:47.643  INFO 1 --- [           main] c.n.d.provider.DiscoveryJerseyProvider   : Using XML decoding codec XStreamXml
2020-10-30 06:39:48.170  INFO 1 --- [           main] c.n.d.s.r.aws.ConfigClusterResolver      : Resolving eureka endpoints via configuration
2020-10-30 06:39:48.294  INFO 1 --- [           main] com.netflix.discovery.DiscoveryClient    : Disable delta property : false
2020-10-30 06:39:48.295  INFO 1 --- [           main] com.netflix.discovery.DiscoveryClient    : Single vip registry refresh property : null
2020-10-30 06:39:48.295  INFO 1 --- [           main] com.netflix.discovery.DiscoveryClient    : Force full registry fetch : false
2020-10-30 06:39:48.295  INFO 1 --- [           main] com.netflix.discovery.DiscoveryClient    : Application is null : false
2020-10-30 06:39:48.295  INFO 1 --- [           main] com.netflix.discovery.DiscoveryClient    : Registered Applications size is zero : true
2020-10-30 06:39:48.295  INFO 1 --- [           main] com.netflix.discovery.DiscoveryClient    : Application version is -1: true
2020-10-30 06:39:48.295  INFO 1 --- [           main] com.netflix.discovery.DiscoveryClient    : Getting all instance registry info from the eureka server
2020-10-30 06:39:48.678  INFO 1 --- [           main] com.netflix.discovery.DiscoveryClient    : The response status is 200
2020-10-30 06:39:48.681  INFO 1 --- [           main] com.netflix.discovery.DiscoveryClient    : Starting heartbeat executor: renew interval is: 30
2020-10-30 06:39:48.739  INFO 1 --- [           main] c.n.discovery.InstanceInfoReplicator     : InstanceInfoReplicator onDemand update allowed rate per min is 4
2020-10-30 06:39:48.745  INFO 1 --- [           main] com.netflix.discovery.DiscoveryClient    : Discovery Client initialized at timestamp 1604039988744 with initial instances count: 3
2020-10-30 06:39:48.747  INFO 1 --- [           main] o.s.c.n.e.s.EurekaServiceRegistry        : Registering application SPRING-CLOUD-MICROSERVICE with eureka with status UP
2020-10-30 06:39:48.748  INFO 1 --- [           main] com.netflix.discovery.DiscoveryClient    : Saw local status change event StatusChangeEvent [timestamp=1604039988748, current=UP, previous=STARTING]
2020-10-30 06:39:48.752  INFO 1 --- [nfoReplicator-0] com.netflix.discovery.DiscoveryClient    : DiscoveryClient_SPRING-CLOUD-MICROSERVICE/spring-cloud-microservice-default-25-7bd5ff7f9b-5426j:spring-cloud-microservice:1025: registering service...
2020-10-30 06:39:48.870  INFO 1 --- [           main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat started on port(s): 1025 (http) with context path ''
2020-10-30 06:39:48.877  INFO 1 --- [           main] .s.c.n.e.s.EurekaAutoServiceRegistration : Updating port to 1025
2020-10-30 06:39:48.954  INFO 1 --- [           main] com.example.demo.DemoApplication         : Started DemoApplication in 20.127 seconds (JVM running for 23.238)
2020-10-30 06:39:49.048  INFO 1 --- [nfoReplicator-0] com.netflix.discovery.DiscoveryClient    : DiscoveryClient_SPRING-CLOUD-MICROSERVICE/spring-cloud-microservice-default-25-7bd5ff7f9b-5426j:spring-cloud-microservice:1025 - registration status: 204
2020-10-30 06:41:51.090  INFO 1 --- [nio-1025-exec-2] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring DispatcherServlet 'dispatcherServlet'
2020-10-30 06:41:51.091  INFO 1 --- [nio-1025-exec-2] o.s.web.servlet.DispatcherServlet        : Initializing Servlet 'dispatcherServlet'
2020-10-30 06:41:51.106  INFO 1 --- [nio-1025-exec-2] o.s.web.servlet.DispatcherServlet        : Completed initialization in 15 ms
2020-10-30 06:43:30.002: No log from server
2020-10-30 06:44:30.006: No log from server
2020-10-30 06:44:48.339  INFO 1 --- [trap-executor-0] c.n.d.s.r.aws.ConfigClusterResolver      : Resolving eureka endpoints via configuration

Regards,
Lalit Kumar

**Comment by jdubois** [comment]
Can you tell exactly which URL you are using? Is this `http://127.0.0.1:8080/hello`? Is it locally or on Azure Spring Cloud? And did you code the controller that should reply to the "hello" endpoint?

**Comment by lalitkumarms** [comment]
Hi,

If take the code from githum then it works fine on local, but if generate the fresh code from the url mentioned to create project it does not work.

Regards,
Lalit kumar
________________________________

**Comment by jdubois** [comment]
Let me rephrase to be sure:
- The sample application on GitHub works
- When you try to do the same, the `http://127.0.0.1:8080/hello` endpoint doesn't work and give you a 404

Are you sure you coded the "hello" endpoint? If you have a 404 it means your server is starting, so it's the controller that is missing.

**Comment by jdubois** [comment]
Closing as this has been inactive for more than 2 months. I've tested everything again in #145 and it definitely works.
